packages/anon-aadhaar-react/src/components/ProveButton.tsx

```diff
@@ -30,31 +31,69 @@ export const ProveButton: React.FC<ProveButtonProps> = ({
 
       if (witness instanceof Error) throw new Error(witness.message)
 
-      const args: AnonAadhaarPCDArgs = {
-        base_message: {
-          argumentType: ArgumentTypeName.BigInt,
-          userProvided: false,
-          value: witness?.msgBigInt.toString(),
-          description: '',
-        },
-        signature: {
-          argumentType: ArgumentTypeName.BigInt,
-          userProvided: false,
-          value: witness?.sigBigInt.toString(),
-          description: '',
-        },
-        modulus: {
-          argumentType: ArgumentTypeName.BigInt,
-          userProvided: false,
-          value: witness?.modulusBigInt.toString(),
-          description: '',
-        },
-        app_id: {
-          argumentType: ArgumentTypeName.BigInt,
-          userProvided: false,
-          value: appId,
-          description: '',
-        },
+      let args: AnonAadhaarPCDArgs
+
+      if (testing === false) {
+        const publicKey = await fetchPublicKey(
+          'https://www.uidai.gov.in/images/authDoc/uidai_offline_publickey_26022021.cer',
+        )
+
+        if (publicKey === null)
+          throw new Error('Error while fetching the public key!')
+
+        args = {
+          base_message: {
+            argumentType: ArgumentTypeName.BigInt,
+            userProvided: false,
+            value: witness?.msgBigInt.toString(),
+            description: '',
+          },
+          signature: {
+            argumentType: ArgumentTypeName.BigInt,
+            userProvided: false,
+            value: witness?.sigBigInt.toString(),
+            description: '',
+          },
+          modulus: {
+            argumentType: ArgumentTypeName.BigInt,
+            userProvided: false,
+            value: '0x' + publicKey,
+            description: '',
+          },
+          app_id: {
+            argumentType: ArgumentTypeName.BigInt,
+            userProvided: false,
+            value: appId,
+            description: '',
+          },
+        }
+      } else {
+        args = {
+          base_message: {
+            argumentType: ArgumentTypeName.BigInt,
+            userProvided: false,
+            value: witness?.msgBigInt.toString(),
+            description: '',
+          },
+          signature: {
+            argumentType: ArgumentTypeName.BigInt,
+            userProvided: false,
+            value: witness?.sigBigInt.toString(),
+            description: '',
+          },
+          modulus: {
+            argumentType: ArgumentTypeName.BigInt,
+            userProvided: false,
+            value: witness?.modulusBigInt.toString(),
```

Comment:
```suggestion
            value: (testing) ? publicKey:  witness?.modulusBigInt.toString(),
```

Reply:
It's elegant but it will fetch the publicKey even when not necessary.
I'll check how to refactor.